src/Tokenizer/Token.php

```diff
@@ -428,6 +428,14 @@ public function isMagicConstant(): bool
         return $this->isArray && isset($magicConstants[$this->id]);
     }
 
+    /**
+     * Check if token is an ampersand.
+     */
+    public function isAmpersand(): bool
```

Comment:
the fact that one has to use `isAmpersand()` instead of `->equals('&')` sounds like bc breaker for end-user.
like, I understand it comes from PHP directly, but I still wonder can we avoid having the effort for each custom fixers to have this upgrade needed

Reply:
how would `PhpToken::is` (ref https://www.php.net/manual/en/phptoken.is.php ) behave for `->is('&')` ?

Reply:
I don't understand how it's BC breaker? No one is forced to change it, keeping `equals('&')` is still... keeping `->equals('&')`, it won't stop to work. For PHP < 8.1. To make it work in PHP 8.1 it has to be changed anyway - to `isGivenKind([T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG, T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG]);` or to `isAmpersand()`.

Reply:
It can be BC breaker for custom Fixers creators if under PHP 7.4 they could use `->equals('&')` and under PHP 8.1 they will need to use `->isAmersand()`

Reply:
Running `->equals('&')` won't work in PHP 8.1 with the already existing code, that the point of this PR.

Adding this method helps with preparing for PHP 8.1 migration, any custom fixer can be refactored to use it, which will change nothing currently, but will make the fixer work in PHP 8.1.

Reply:
I would rather make a `if (... === '&')` inside the `equals` method to keep BC for custom fixers

Reply:
I still don't understand how thus suppose to be BC break. This is only extracting it to separate method and will be extend when we extend support to include PHP 8.1.

Reply:
Adding new method is not a BC breaker.

the BC breaker is for libraries adding new rules - nowadays when one wants to compare with `&`, they can use `->equals('&')`. With upcoming changes in PHP , one will need to do `$this->isGivenKind([T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG, T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG])` and their `->equals('&')` will no longer work.

adding a proxy method, while handy, is not solving the BC breaker for custom rules creators - they will still need to change their existing codebases to stop using `->equals('&')`. The root cause is coming from php-src and not our repo, but for end user it's a bc breaker - rule working on one PHP version will stop working on other PHP version. Right now, we prevent that by setting up max allowed PHP version. You want to prepare to bump allowed max PHP version , but current proposal will lead to issues to custom rules creators.